src/ODataHttpClient/Models/Request.cs

```diff
@@ -73,9 +73,14 @@ public static Request Create<T>(HttpMethod method, string uri, T body, IEnumerab
 
             var type = typeof(T);
 
-            if (type.IsValueType || type == stringType)
+            if (type == stringType)
             {
-                content = body.ToString();
+                content = body.ToString(); 
+                mime = "text/plain";
+            }
+            else if (type.IsValueType)
+            {
+                content = serializer.Serialize<T>(body);
```

Comment:
This request content type is needed plain text whichi cannnot use serializer. use `ToString` with CultureInfo.InvariantCulture.